build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set(TARGET_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(DOWNLOAD_SCRIPT_URL "https://raw.githubusercontent.com/NVIDIA/build-system-archive-import-examples/main/parse_redist.py")
+set(DOWNLOAD_DIR ${TARGET_DIR}/download)
+set(DOWNLOAD_SCRIPT_PATH ${DOWNLOAD_DIR}/parse_redist.py)
+
+# Parameters to the download script.
+# Look for an appropriate redistrib_*.json here to verify:
+#   https://developer.download.nvidia.com/compute/cuda/redist/
+set(VERSION "11.6.2")
+set(PRODUCT "cuda")
+if(UNIX)
+  set(OS "linux")
+elseif(WIN32)
+  set(OS "windows")
+else()
+  message(SEND_ERROR "Unsupported OS environment. Must be Windows or Unix.")
```

Comment:
Is this linux or unix? Does the linux one work on mac?

We should call out that the OS and arch stuff is being hardcoded and we should probably be detecting arch, for instance. Not saying that it's necessarily worth it to try to detect it here right now (as apparently CMake doesn't have a builtin way to do that), but especially if we start actually supporting Mac M1s, we'll need to fix all this to be more flexible. This would also mean that we should at least set the default for CUDA to off for MacOS

Reply:
Agreed, changed to "Linux". Also I believe CUDA isn't supported on Macs so really only Linux and Windows are in play here.

Reply:
Yeah I just meant you should add a note.